onnxruntime/core/optimizer/propagate_cast_ops.h

```diff
@@ -17,14 +18,18 @@ class PropagateCastOps : public GraphTransformer {
  public:
   PropagateCastOps(GraphTransformerConfiguration::PropagateCastOpsConfiguration::Strategy strategy =
                        GraphTransformerConfiguration::PropagateCastOpsConfiguration::Strategy::FloodFill,
-                   size_t level = 0, const std::vector<std::string>& allow_list = {},
-                   const std::unordered_set<std::string>& compatible_execution_providers = {}) noexcept;
+                   size_t level = 0, const gsl::span<std::string const>& allow_list = {},
+                   const std::unordered_set<std::string>& compatible_execution_providers = {});
+
+  ~PropagateCastOps() noexcept;
 
   Status ApplyImpl(Graph& graph, bool& modified, int graph_level, const logging::Logger& logger) const override;
 
+  ORT_DISALLOW_COPY_AND_ASSIGNMENT(PropagateCastOps);
+
  private:
-  size_t level_;
-  GraphTransformerConfiguration::PropagateCastOpsConfiguration::Strategy strategy_;
+  class Impl;
+  Impl* impl_;
```

Comment:
1. Curious why another level of indirection is needed.
2. If needed, any reason to not use unique_ptr?

Reply:
The reason being here, is that I had to make some static functions members because they used to operate on global data structures that are not global anymore. Those functions require headers that I did not want to bring into this header.